packages/color-scale-block/src/ColorScaleBlock.tsx

```diff
@@ -0,0 +1,481 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import 'tailwindcss/tailwind.css';
+import { MouseEvent, useEffect, useRef, useState } from 'react';
+import {
+    AppBridgeBlock,
+    FrontifyColorPalette,
+    useBlockSettings,
+    useColorPalettes,
+    useEditorState,
+} from '@frontify/app-bridge';
+import { SquareWithColor } from './components/SquareWithColor';
+import { ColorPickerFlyout } from './components/ColorPickerFlyout';
+import { ColorProps, Settings } from './types';
+import {
+    Button,
+    ButtonGroup,
+    ButtonSize,
+    ButtonStyle,
+    DropZonePosition,
+    IconArrowStretchBox12,
+    IconPlus12,
+    Palette,
+    useMemoizedId,
+} from '@frontify/fondue';
+import '@frontify/fondue-tokens/styles';
+import { DndProvider } from 'react-dnd';
+import { HTML5Backend } from 'react-dnd-html5-backend';
+import 'tailwindcss/tailwind.css';
+import { DropZone } from './dragAndDrop/DropZone';
+import { EmptyView } from './components/EmptyView';
+import {
+    calculateDefaultColorWidth,
+    calculateWidths,
+    canExpandColorBlock,
+    defaultColorSquareWidth,
+    resizeEvenly,
+} from './helpers';
+import { MouseEventHandler } from 'react';
+
+type Props = {
+    appBridge: AppBridgeBlock;
+};
+
+export const ColorScaleBlock = ({ appBridge }: Props) => {
+    const { colorPalettes } = useColorPalettes(appBridge);
+    const [colorPickerPalette, setColorPickerPalette]: [Palette[], (color: Palette[]) => void] = useState(
+        [] as Palette[]
+    );
+    const isEditing = useEditorState(appBridge);
+    const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
+    const [colorScaleHeight, setColorScaleHeight] = useState(
+        blockSettings['customHeight'] ? blockSettings['heightInput'] : blockSettings['heightSlider']
+    );
+    const [currentlyDraggedColorId, setCurrentlyDraggedColorId]: [
+        number | null | undefined,
+        (value?: number | null | undefined) => void
+    ] = useState();
+    const [editedColor, setEditedColor]: [
+        ColorProps | null | undefined,
+        (color: ColorProps | null | undefined) => void
+    ] = useState<Nullable<ColorProps>>();
+    const [isColorPickerOpen, setIsColorPickerOpen] = useState(false);
+    const colorScaleBlockRef: { current: HTMLDivElement | null } = useRef(null);
+    const draggingId: { current?: number | null | undefined } = useRef(null);
+    const dragStartPos: { current?: number | null | undefined } = useRef();
+    const dragStartWidth: { current?: number | null | undefined } = useRef();
+    const lastDragPos: { current?: number | null | undefined } = useRef();
+    const timerToUpdateBlockSettings: { current?: ReturnType<typeof setTimeout> | undefined } = useRef(undefined);
+    const minimumNumberOfColors = 1;
+    const maximumNumberOfPlaceholderSquares = 6;
+
+    const deleteColor = (id: number | undefined) => {
```

Comment:
Shouldn't be able to call this method on color without id
```suggestion
    const deleteColor = (id: number) => {
```